Allow non-power-of-two directional shadow size in 3D

Calinou opened the pull request:
`master` version of https://github.com/godotengine/godot/pull/54042.

This allows for more finegrained performance/quality tuning, especially at higher sizes. Any number can be used. For directional light shadows, 3072 and 6144 are sensible shadow sizes that can be used if the default 4096 is too slow or too low-resolution for a large scene.

## Preview

![image](https://github.com/user-attachments/assets/7ed16b5e-de59-42c3-9318-00745d74001c)

Calinou commented:
> Nice change, I can totally see this being useful, makes me wonder what would happen if they were uncapped (so going past 16k) when going past the current limit.. thinking

GPUs don't support textures larger than 16384×16384, so there's no point. Not to mention a 16384×16384 shadowmap is pretty slow already.

Calinou commented:
I amended this PR to only allow NPOT sizes for directional shadows, not point light shadows.

reduz commented:
@Calinou I think using custom sizes is still not a great idea because for users its difficult to change this value. IMO I would simply do an enum with more intermediate values (2^n + 2^(n-1) ) such as:

* 256
* 384
* 512
* 768
* 1024
* 1536
* 2048
* 3072
* 4096
* 6144
* 8192

This is considerably easier for users to edit and it still provides relatively good divisors.

Calinou commented:
> This is considerably easier for users to edit and it still provides relatively good divisors.

See also https://github.com/godotengine/godot/pull/57610, where a counterargument to using enusm was made.

Calinou commented:
Since https://github.com/godotengine/godot/pull/57610 was closed as it broke compatibility, this means we can't go with reduz's suggestion unless we break compatibility or implement a compatibility handler. This would mean creating a new project setting with a different name, hiding the old project setting and reading the old project setting's name on startup to set the new value.

Calinou commented:
> I think going with reduz's suggestion along with a compatibility handler would be fine. IMO having Po2 or near-Po2 textures is ideal as it makes having multiple shadows much simpler. If we allow non-Po2 we will run into aliasing issues when the atlas doesn't neatly subdivide by the number of DirectionalLights that we have

I assume a compatibility handler requires renaming the project setting, so that we can use a magic getter to read the old property name?

Calinou committed:
Allow non-power-of-two directional shadow size in 3D

This allows for more finegrained performance/quality tuning, especially
at higher sizes. Any number can be used. For directional light shadows,
3072 and 6144 are sensible shadow sizes that can be used if the
default 4096 is too slow or too low-resolution for a large scene.

Calinou commented:
Rebased and tested again, it works as expected.

This now keeps the integer property and uses an enum hint with numbered values, so that compatibility is preserved. See OP for an updated screenshot.
